crates/ruff_python_parser/src/semantic_errors.rs

```diff
@@ -25,6 +25,8 @@ pub struct Checkpoint {
 
 #[derive(Debug, Default)]
 pub struct SemanticSyntaxChecker {
+    source_type: PySourceType,
```

Comment:
Should this be a method on `ctx` similar to `PythonVersion` because it isn't state?